src/lib/vm-manager-hoc.jsx

```diff
@@ -51,29 +48,26 @@ const vmManagerHOC = function (WrappedComponent) {
                     this.props.onLoadedProject(this.props.loadingState, this.props.canSave);
                 })
                 .catch(e => {
-                    // Need to catch this error and update component state so that
-                    // error page gets rendered if project failed to load
-                    this.setState({loadingError: true, errorMessage: e});
+                    this.props.onError(`vm-manager-hoc error: ${e}`);
```

Comment:
Did you add this context because we don't have a good stack trace with this way of error handling? Maybe we shouldn't be setting state to the stringified error, but the full error object, so it can contain a stack trace? (If it already stringifies with a stack trace then I would say the added context is unnecessary.)

Reply:
Just realized it had been passing the full error. I think we should continue doing that, and expect the error set with `onError` to be an error object, not a string. As a follow up we should start outputting `e.stack` to get the stack trace.

Reply:
to be clear, for now I'll replace the `errorMessage` prop entirely with an `error` prop, and only get the string from that when outputting it to the user

Reply:
But, there's a bigger issue. I would like to include context that we can display to the user, for help in practically understanding where errors occur, both when they happen to users and to us. Maybe we could pass *both* a string reflecting where the error occured, and the error object itself?

Reply:
that sounds good. i'll make a follow on issue.

Reply:
Made https://github.com/LLK/scratch-gui/issues/3468